ecommerce/views.py

```diff
@@ -337,6 +341,53 @@ def post(self, request, *args, **kwargs):  # noqa: ARG002
             },
         )
 
+    def put(self, request):
+        """
+        Deactivate coupon(s) whose coupon code(s) or name(s) are passed in the request body,
+        with each entry separated by a new line.
+        """
+        coupon_codes_and_payment_names = request.data.get("coupons", "").split("\n")
```

Comment:
I asked you to remove the strip, but it turns out that I can pass multiple blank lines; in that case, it will display blank as skipped. You can bring back the strip. Thanks!

Reply:
Changing the code to
```suggestion
        coupon_codes_and_payment_names = [code_and_payment for code_and_payment in request.data.get("coupons", "").strip().split("\n")if code_and_payment]
```
Consider this as input to the api: `"abc\n\n\nanc"`. strip will not remove the extra line and split("\n") will add an empty string in the list.